krabs/krabs/filtering/event_filter.hpp

```diff
@@ -107,8 +116,13 @@ namespace krabs {
     : predicate_(predicate)
     {}
 
-	inline event_filter::event_filter(unsigned short event_id, filter_predicate predicate/*=nullptr*/)
-    : provider_filter_event_id_(event_id),
+    inline event_filter::event_filter(std::vector<unsigned short> event_ids, filter_predicate predicate/*=nullptr*/)
+    : provider_filter_event_ids_(event_ids.begin(), event_ids.end()),
```

Comment:
In this case, prefer `provider_filter_event_ids_{ event_ids }` which will invoke the vector copy ctor.